engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -580,31 +587,7 @@ class JoinPartitionKernel : public kernel {
 			}
 		}
 
-		auto nodes = local_context->getAllNodes();
```

Comment:
leave this logic where it currently lies just take out the repetitive stuff that actually generates the metadata.

Reply:
One of the key ideas behind the message manager is that it maintains the state necessary for proper sending and receiving of messages such as the node_count, or PARTITION_COUNT. this communication Paradigm relies on certain usage patterns. To make it safer to use it's better to encapsulate the usage patterns in function calls

Reply:
What I dont get is that the logic is rather simple before. Send a list of messages, wait for their counterparts to arrive. The only thing that was really repetitive was the generate of the message metadata. I am saying this class feels a bit pointless